src/modules/core/private-storage/controllers/private-storage.controller.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  Body,
+  Controller,
+  HttpStatus,
+  Inject,
+  Post,
+  UseGuards,
+} from '@nestjs/common';
+import {
+  ApiBearerAuth,
+  ApiCreatedResponse,
+  ApiOperation,
+  ApiTags,
+} from '@nestjs/swagger';
+import { CustomApiResponse } from '@src/decorators/custom-api-response.decorator';
+import { JwtAuthGuard } from '@src/guards/jwt-auth.guard';
+import { GetSignedUrlDto } from '../dtos/get-signed-url.dto';
+import {
+  PrivateStorageService,
+  PRIVATE_STORAGE_SERVICE,
+} from '../interfaces/private-storage-service.interface';
+import { GetSignedUrlResponseType } from '../types/response/get-signed-url-response.type';
+
+@ApiTags('PrivateStorage')
+@Controller('api/private-storage')
+export class PrivateStorageController {
+  constructor(
+    @Inject(PRIVATE_STORAGE_SERVICE)
+    private privateStorageService: PrivateStorageService,
+  ) {}
+
+  @Post('/signedUrl')
```

Comment:
```suggestion
  @Post('/signed-url')
```
url 은 카멜케이스가 아닌 -로 공백을 표현해주세요
그리고 POST 메서드를 사용하는게 맞는지 검토해볼 필요가 있을거같아요 실제로 우리 서버에 리소스를 생성하는게 아닌 미리 서명된 url 을 가져온다는 의미가 맞아서 개인적으로는 GET method 가 맞다고 생각되는데 @RokwonK 님은 어떻게 생각하시나요???

Reply:
케밥케이스로 수정하도록 하겠습니다!

제가 post 요청으로 해당 api를 만든 이유에 대해서 설명 드리자면
1. get 요청에 request body를 사용하는 것이 그렇게 할 수는 있지만 일반적이지 못하다고 판단하였습니다.
2. 서버에 리소스를 생성하진 않았지만 s3에 접근할 수 있는 권한을 가진 url을 생성한 것이니 post가 적합하다고 판단하였습니다!
 다른 분들 생각은 어떠신가요??

Reply:
- url을 - 로 공백 표시하는 이유가 있나요?
- 저도 Get 요청에 가깝다고 생각합니다 클라이언트 측에서 데이터를 저장하는게 아니라 가져오는 거에 더 가깝다고 생각합니다!

Reply:
@RokwonK  케밥케이스가 일반적으로 널리 사용되고 있고, 구글 검색 엔진 관련 문서에도 케밥 케이스를 사용하라고 권장하고 있더라구요!

https://developers.google.com/search/docs/crawling-indexing/url-structure?hl=en&visit_id=638047813916368146-1007854177&rd=1

Reply:
메서드랑 클라이언트에게 받을 데이터 어디서 받을지 결정하면 공유 한번 해주세여

Reply:
여러분~~  HTTP 메서드 방식 결정하기 이전에 한번 더 의견 여쭙고 싶어서 댓글 작성함다..

이런 저런 자료를 많이 찾아봤는데 post 요청이 더 적합하다고 생각이 드는데 그 이유는 위에서도 말씀드렸지만 클라이언트 단에서 서버에 리소스를 생성하진 않았지만 클라이언트의 요청을 통해 S3에 접근할 수 있는 권한을 가진 url을 생성 한 것이니 post가 적합하다고 생각 합니다. 그리고 많은 예제가 post로 구현되어 있더라구요!

록원님과 석호님 말씀처럼 get요청으로 바꾸게 된다면 query string으로 데이터를 받는 방향으로 수정하겠습니다!
여러 글을 찾아보니 클라이언트측의 특정 http라이브러리에서 get요청에 body를 허용하지 않는 라이브러리가 있는것 같더라구요!

관련 자료를 아래에 링크 걸어놓겠습니다 한번 들어가셔서 보셔요!

https://if1live.github.io/posts/http-get-request-with-body-and-http-library/

Reply:
사실 Post 도 어느정도 맞다고 생각됩니다.
post 로 갈 갈 경우 data 를 body 에 받을 수 있는 이점이 있고 캐싱이 안된다는 단점이 있습니다.
get 으로 갈 경우 사실 body 를 제외하면 데이터를 받을 위치가 애매하긴 하네여 get 으로 만들 경우 http 프로토콜상에서 캐싱이 가능하다는 이점이 있는데 이 api 경우 캐싱이 필요없기때문에 post로 해도 상관 없을거같아요

Reply:
아하 get 요청으로 바꾼다면 query string이 맞다고 생각합니다.
클라이언트측에서 요청 할 때는 서버내부동작(signed url 발급)을 아는 것은 아니고 단지 '허용 url을 받아온다' 라는 동작이기 때문에 GET이 적합하다 생각 하지만 보안상 이유로 쿼리노출이 아닌 body에 포함(POST  + TSL로 보안 up)해야한다면 post로 요청도 적합하다고 생각해요!

Reply:
@rrgks6221 , @RokwonK  좋은 의견 감사드립니다!